InvenTree/plugin/builtin/barcodes/mouser_barcode.py

```diff
@@ -0,0 +1,92 @@
+"""The MouserBarcodePlugin matches Mouser barcodes to supplier parts."""
+
+import logging
+
+from django.utils.translation import gettext_lazy as _
+
+from company.models import ManufacturerPart, SupplierPart
+from plugin import InvenTreePlugin, registry
+from plugin.mixins import BarcodeMixin, SettingsMixin
+from .barcode2d import parse_ecia_barcode2d
+
+logger = logging.getLogger('inventree')
+
+
+class MouserBarcodePlugin(BarcodeMixin, SettingsMixin, InvenTreePlugin):
+    """BarcodePlugin for matching Mouser barcodes."""
+
+    NAME = "MouserBarcode"
+    TITLE = _("Mouser Barcodes")
+    DESCRIPTION = _("Provides support for scanning Mouser barcodes")
+    VERSION = "1.0.0"
+    AUTHOR = _("InvenTree contributors")
+
+    SETTINGS = {
+        "MOUSER_SUPPLIER_NAME": {
+            "name": _("Mouser Supplier Name"),
+            "description": _("Name of the Supplier which acts as 'Mouser'"),
+            "default": "Mouser",
+            "validator": lambda name: bool(name),
```

Comment:
Use a model field like this:
```suggestion
            "validator": lambda name: bool(name),
            "model": "company.company"
```

Reply:
Yep, if you use a "model" attribute you can allow the user to select the supplier directly. So, you don't need to have them input a name and use name lookup (and then potentially change the config if the company name ever changes). They can just select "Mouser" (e.g.) from a drop-down list of companies in the system.

Reply:
Thanks! I've been looking for something like that for a solid 15 minutes yesterday, but didn't find anything ^^

Reply:
I've also added back some code from an ealier revision which tries to automatically guess the Mouser supplier if it hasn't been set yet (I feel like that's a good QOL addition).